### PR TITLE
Add plugin attachmentsid

### PR DESCRIPTION
**Pull request by JLTRY**
{attachmentsid id=xx,yy}
where xx,yy.. are the id(s) of the attachment(s)

inspired by https://github.com/jmcameron/attachments/pull/3

### CODE REVIEW

**Comment by JLTRY** [comment]
I keep id=xxx as parameter as we could also add other parameters to customize the view

**Comment by parapente** [comment]
I think that we don't really need another plugin for this extension @JLTRY but we should incorporate the id= functionality in the attachments framework plugin. I have a partial implementation that seems to work but not correctly right now. I believe that I will have a draft by tomorrow that I will add to this PR if you don't mind.

However I believe that for the functionality to be complete we should add another button in the editor that would allow you to select the attachment from a list as it would be difficult for most people to know the id of the attachment.

**Comment by JLTRY** [comment]
Agree with you for keeping same plugin. To be noticed that mine is working. For button I have got a code that works with the choice of existing attachments, and I will make a pull request soon

File: attachments_component/site/src/Model/AttachmentsModel.php [hunk 1]
```diff
@@ -387,15 +391,19 @@ public function &getAttachmentsList()
...
+			}
 		if ( $parent_id == 0 ) {
 			// If the parent ID is zero, the parent is being created so we have
 			// do the query differently
 			$user_id = $user->get('id');
 			$query->where('a.parent_id IS NULL AND u.id=' . (int)$user_id);
 			}
 		else {
-			$query->where('a.parent_id='.(int)$parent_id);
+			if ($parent_id != '%') {
```
Comment (parapente) [inline attachments_component/site/src/Model/AttachmentsModel.php:404]: @JLTRY I cannot understand what you want to achieve here. I can see that you set parent_id to % earlier but since you check if parent isn't % why did you change the query from "a.parent_id=" to "a.parent_id like"? Did you want to add attachments from other articles maybe?

File: attachments_component/site/src/Controller/AttachmentsController.php [hunk 2]
```diff
@@ -89,9 +90,12 @@ public function displayString($parent_id, $parent_type, $parent_entity,
 		$model = $this->getModel('Attachments', 'Site');
 		if ( !$model ) {
 			$errmsg = Text::_('ATTACH_ERROR_UNABLE_TO_FIND_MODEL') . ' (ERR 60)';
-			throw new \Exception($errmsg, 500);
```
Comment (parapente) [inline attachments_component/site/src/Controller/AttachmentsController.php:92]: Why did you remove the exception here? Did you find a bug that something wasn't handled correctly?

File: attachments_component/site/src/Controller/AttachmentsController.php [hunk 3]
```diff
@@ -89,9 +90,12 @@ public function displayString($parent_id, $parent_type, $parent_entity,
...
+			//throw new \Exception($errmsg, 500);
+			return;
 			}
-
+		if ($attachmentsid) {
+			$parent_id = '?';
```
Comment (parapente) [inline attachments_component/site/src/Controller/AttachmentsController.php:97]: Why do you change parent_id to ?

File: attachments_component/site/src/Controller/AttachmentsController.php (hunk 2 above)
Comment (JLTRY) [reply attachments_component/site/src/Controller/AttachmentsController.php:92]: hello sorry yes I should put it back

File: attachments_component/site/src/Controller/AttachmentsController.php (hunk 3 above)
Comment (JLTRY) [reply attachments_component/site/src/Controller/AttachmentsController.php:97]: the idea was to be able to select other attachments as current article

File: attachments_component/site/src/Model/AttachmentsModel.php (hunk 1 above)
Comment (JLTRY) [reply attachments_component/site/src/Model/AttachmentsModel.php:404]: yes sorry the query it should not be changed for a.parent_id. Yes if attachmentsid is not null it is of format xx,yy,zz. YEs I want to be able to add attachments from everywhere

**Comment by parapente** [comment]
I have pushed the changes to make attachments framework plugin handle {attachments id=xx,yy,zz} style tokens. It doesn't handle attachments from other parents yet but can be modified to handle this use case if needed. Let's start small and expand later. I believe that most of the times the user would want to add all the files to the article and then add the needed file links in specific positions in the article. If we add attachments from other parents we might want to check if the parent is disabled/recycled/archived to allow adding/showing the attachment which will make things a bit more complicated.

**Comment by JLTRY** [comment]
I corrected some issue for attachmentsid

**Comment by parapente** [comment]
@JLTRY the changes I pushed were for using just the original plugins with {attachments id=xxx,yyy} so that is why it broke the new plugin. Also it is not really safe to get text from the user and pass it as is to the query. That is why I used a regex that expects only digits and the comma character next to "id=". Try disabling attachmentsid plugin and use the token. It should work as is.

**Comment by parapente** [comment]
Also I forgot a couple of log messages in the code that we should remove before merging.

**Comment by JLTRY** [comment]
Now it should work and I removed the log
